hoomd/md/methods.py

```diff
@@ -475,21 +499,131 @@ class nph(NPT):
+    def __init__(self, filter, tau, S, tauS, couple, box_dof=[True,True,True,False,False,False], rescale_all=False, gamma=0.0):
+        # store metadata
+        param_dict = ParameterDict(
+            filter=ParticleFilter,
+            kT=Variant,
+            tau=float(tau),
+            S=OnlyIf(to_type_converter((Variant,)*6), preprocess=self.__preprocess_stress),
+            tauS=float(tauS),
+            couple=str(couple),
+            box_dof=(bool,)*6,
+            rescale_all=bool(rescale_all),
+            gamma=float(gamma),
+            translational_thermostat_dof=(float, float),
+            rotational_thermostat_dof=(float, float),
+            barostat_dof=(float, float, float, float, float, float))
 
-        # initialize base class
-        npt.__init__(self, nph=True, kT=1.0, **params)
+        param_dict.update(
+            dict(filter=filter,
+                 kT=hoomd.variant.Constant(1.0),
+                 S=S,
+                 couple=couple,
+                 box_dof=box_dof,
+                 translational_thermostat_dof=(0, 0),
+                 rotational_thermostat_dof=(0, 0),
+                 barostat_dof=(0, 0, 0, 0, 0, 0)))
+
+        # set defaults
+        self._param_dict.update(param_dict)
+
+    def _attach(self):
+        # initialize the reflected c++ class
+        if isinstance(self._simulation.device, hoomd.device.CPU):
+            cpp_cls = _md.TwoStepNPTMTK
+            thermo_cls = _md.ComputeThermo
+        else:
+            cpp_cls = _md.TwoStepNPTMTKGPU
+            thermo_cls = _md.ComputeThermoGPU
+
+        cpp_sys_def = self._simulation.state._cpp_sys_def
+        thermo_group = self._simulation.state._get_group(self.filter)
+
+        thermo_half_step = thermo_cls(cpp_sys_def,
+                                      thermo_group,
+                                      "")
+
+        thermo_full_step = thermo_cls(cpp_sys_def,
+                                      thermo_group,
+                                      "")
+
+        self._cpp_obj = cpp_cls(cpp_sys_def,
+                                thermo_group,
+                                thermo_half_step,
+                                thermo_full_step,
+                                self.tau,
+                                self.tauS,
+                                self.kT,
+                                self.S,
+                                self.couple,
+                                self.box_dof,
+                                True)
+
+        # Attach param_dict and typeparam_dict
+        super()._attach()
+
+    def __preprocess_stress(self, value):
+        if isinstance(value, Sequence):
+            if len(value) != 6:
+                raise ValueError(
+                    "Expected a single hoomd.variant.Variant / float or six.")
+            return tuple(value)
+        else:
+            return (value, value, value, 0, 0, 0)
 
     def randomize_velocities(self, kT, seed):
```

Comment:
Remove this method. 

Copy `thermalize_thermostat_and_barostat_dof` from NPT, but rename to `thermalize_barostat_dof` update the documentation to remove reverences to the thermostat. Note that the C++ method is still called `thermalizeThermostatAndBarostatDOF` - it will ignore the thermostat in NPH mode.